### PR TITLE
Return targets created by the user

### PR DESCRIPTION
**Pull request by Ricoch**
Tasks:
- Renamed "_target.json.jbuilder" to "_basic.json.jbuilder" for better clarity when having different ways of returning the target
- Added index.json.jbuilder file to manage the index action response
- Added index to target routes
- Added tests for index action

### CODE REVIEW

File: spec/requests/api/v1/targets/index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+describe 'GET /api/v1/targets/', type: :request do
+  let(:user)         { create(:user) }
+  let(:targets_path) { '/api/v1/targets' }
+  let!(:targets)     { create_list(:target, 3, user: user) }
+
+  describe 'GET index' do
```
Comment (BelenRemedi) [inline spec/requests/api/v1/targets/index_spec.rb]: Maybe this `describe` can be omitted.

File: spec/requests/api/v1/targets/index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    context 'being signed in' do
+      let(:first_actual_target) { json[:targets].first }
+
+      it 'returns success' do
+        get api_v1_targets_path, headers: auth_headers, as: :json
+
+        expect(response).to have_http_status(:success)
+      end
+
+      it 'returns user\'s targets' do
```
Comment (BelenRemedi) [inline spec/requests/api/v1/targets/index_spec.rb]: Why this `\`?

File: spec/requests/api/v1/targets/index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+describe 'GET /api/v1/targets/', type: :request do
+  let(:user)         { create(:user) }
+  let(:api_v1_targets_path) { '/api/v1/targets' }
+  let!(:targets)     { create_list(:target, 3, user: user) }
+
+  describe 'GET index' do
```
Comment (BelenRemedi) [inline spec/requests/api/v1/targets/index_spec.rb]: Maybe this `describe` can be omitted.

File: spec/requests/api/v1/targets/index_spec.rb (hunk 2 above)
Comment (Ricoch) [reply spec/requests/api/v1/targets/index_spec.rb]: It is so I can use the apostrophe for the text "user's"

**Review by BelenRemedi — APPROVED** [review]

**Review by letiesperon — APPROVED** [review]

**Pull request merged**